subsys/lorawan/services/frag_decoder_lowmem.c

```diff
@@ -0,0 +1,411 @@
+/*
+ * Copyright (c) 2024 A Labs GmbH
+ * Copyright (c) 2024 tado GmbH
+ * Copyright (c) 2022 Jiapeng Li
+ *
+ * Based on: https://github.com/JiapengLi/LoRaWANFragmentedDataBlockTransportAlgorithm
+ *
```

Comment:
Probably worth including a link to the original algorithm paper linked in the docs: http://www.inference.org.uk/mackay/gallager/papers/ldpc.pdf